hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/OzoneStreamCapabilities.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.ozone;
+
+import org.apache.hadoop.classification.InterfaceAudience;
+import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.hadoop.fs.StreamCapabilities;
+/**
+ * Interface to query streams for supported capabilities of Ozone.
+ *
+ * Capability strings must be in lower case.
+ *
+ * Constant strings are chosen over enums in order to allow other file systems
+ * to define their own capabilities.
+ */
+@InterfaceAudience.Public
+@InterfaceStability.Evolving
+public interface OzoneStreamCapabilities extends StreamCapabilities {
+
+
+  /**
+   * Stream read(ByteBuffer) capability implemented by
+   * {@link ByteBufferReadable#read(java.nio.ByteBuffer)}.
+   */
+  String READBYTEBUFFER = "in:readbytebuffer";
+
+
+  /**
+   * Capabilities that a stream can support and be queried for.
+   */
+  @Deprecated
+  enum StreamCapability {
+    READBYTEBUFFER(OzoneStreamCapabilities.READBYTEBUFFER);
+    private final String capability;
+
+    StreamCapability(String value) {
+      this.capability = value;
+    }
+
+    public final String getValue() {
+      return capability;
+    }
+  }
+
+  /**
+   * Query the stream for a specific capability.
+   *
+   * @param capability string to query the stream support for.
+   * @return True if the stream supports capability.
+   */
+  boolean hasCapability(String capability);
```

Comment:
No need to duplicate the method from parent interface.

Reply:
Thanks @adoroszlai for reviewing.

I would violate the ```checkstyle``` with ```[ERROR] src/main/java/org/apache/hadoop/fs/ozone/OzoneStreamCapabilities.java:[30] (design) InterfaceIsType: interfaces should describe a type and hence have methods.```  when removing the method.

I think the we could disable the ```InterfaceIsType``` by removing this [line](https://github.com/apache/hadoop-ozone/blob/master/hadoop-hdds/dev-support/checkstyle/checkstyle.xml#L180), but I don't know whether it is proper to do this.

How do you think about this ?

Reply:
Nice catch.  In that case I think it can be a class (only for holding the constant), referencing the Hadoop `StreamCapabilities` interface in javadoc.  Or we can drop the idea and define the constant directly in `OzoneFSInputStream` (in that case sorry about the dead-end suggestion).

Reply:
Thanks @adoroszlai for helping and reviewing it.
I converted ```OzoneStreamCapabilities``` to utility class, and updated related string with it.